tutorials/general/t8_step7_interpolation.cxx

```diff
@@ -0,0 +1,421 @@
+    }
+  }
+  /* Old element is coarsened */
+  else if (refine == -1) {
+    adapt_data_new->element_data[first_incoming].values = 0;
+    for (t8_locidx_t i = 0; i < num_outgoing; i++) {
+      adapt_data_old->element_data[first_outgoing+i].values;
+          adapt_data_new->element_data[first_incoming].values +=
+            adapt_data_old->element_data[first_outgoing+i].values;
+    }
+    adapt_data_new->element_data[first_incoming].values /= num_outgoing;
+  }
+}
+
+/* Each process has computed the data entries for its local elements.
+ * In order to get the values for the ghost elements, we use t8_forest_ghost_exchange_data.
+ * Calling this function will fill all the ghost entries of our element data array with the
+ * value on the process that owns the corresponding element. */
+static void
+t8_step7_exchange_ghost_data (t8_forest_t forest,
+                              struct t8_step7_element_data_t *data)
+{
+  sc_array           *sc_array_wrapper;
+  t8_locidx_t         num_elements =
+    t8_forest_get_local_num_elements (forest);
+  t8_locidx_t         num_ghosts = t8_forest_get_num_ghosts (forest);
+
+  /* t8_forest_ghost_exchange_data expects an sc_array (of length num_local_elements + num_ghosts).
+   * We wrap our data array to an sc_array. */
+  sc_array_wrapper =
+    sc_array_new_data (data, sizeof (struct t8_step7_element_data_t),
+                       num_elements + num_ghosts);
+
+  /* Carry out the data exchange. The entries with indices > num_local_elements will get overwritten. */
+  t8_forest_ghost_exchange_data (forest, sc_array_wrapper);
+
+  /* Destroy the wrapper array. This will not free the data memory since we used sc_array_new_data. */
+  sc_array_destroy (sc_array_wrapper);
+}
+
+/* Write the forest and the data corresponding to the forest in a vtu file. 
+ *
+ * \param [in] forest           Forest that should written in the vtu file
+ * \param [in] data             Data corresponding to the forest
+ * \param [in] prefix           Prefix to define the file name
+ */
+static void
+t8_write_vtu (t8_forest_t forest,
+                struct t8_step7_adapt_data *data,
+                const char *prefix)
+{
+  t8_locidx_t         num_elements =
+    t8_forest_get_local_num_elements (forest);
+  t8_locidx_t         ielem;
+  /* We need to allocate a new array to store the volumes on their own.
+   * This array has one entry per local element. */
+  double             *element_data = T8_ALLOC (double, num_elements);
+  /* The number of user defined data fields to write. */
+  int                 num_data = 1;
+  t8_vtk_data_field_t vtk_data;
+  vtk_data.type = T8_VTK_SCALAR;
+  strcpy (vtk_data.description, "Element own data");
+  vtk_data.data = element_data;
+  /* Copy the elment's data from the data array to the output array. */
+  for (ielem = 0; ielem < num_elements; ++ielem) {
+    element_data[ielem] = (data->element_data[ielem]).values;
+  }
+
+  {

```

Comment:
Is there a reason to put this in a seperate block?
This style confused me a little.